### PR TITLE
[Streaming][DOC] Fix typo & formatting for JavaDoc

### PR DESCRIPTION
**Pull request by niofire**
## What changes were proposed in this pull request?
- Fixed typo for function outputMode
      - OutputMode.Complete(), changed `these is some updates` to `there are some updates`
- Replaced hyphenized list by HTML unordered list tags in comments to fix the Javadoc documentation.

Current render from most recent [Spark API Docs](https://spark.apache.org/docs/2.3.1/api/java/org/apache/spark/sql/streaming/DataStreamWriter.html):

#### outputMode(OutputMode) - List formatted as a prose.

image

#### outputMode(String) - List formatted as a prose.
image

#### partitionBy(String*) - List formatted as a prose.
image

## How was this patch tested?
This PR contains a document patch ergo no functional testing is required.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
ok to test

File: sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala [hunk 1]
```diff
@@ -46,14 +46,16 @@ final class DataStreamWriter[T] private[sql](ds: Dataset[T]) {
...
   /**
    * Specifies how data of a streaming DataFrame/Dataset is written to a streaming sink.
-   *   - `OutputMode.Append()`: only the new rows in the streaming DataFrame/Dataset will be
-   *                            written to the sink
-   *   - `OutputMode.Complete()`: all the rows in the streaming DataFrame/Dataset will be written
-   *                              to the sink every time these is some updates
-   *   - `OutputMode.Update()`: only the rows that were updated in the streaming DataFrame/Dataset
+   * <ul>
+   *   <li> `OutputMode.Append()`: only the new rows in the streaming DataFrame/Dataset will be
+   *                            written to the sink.</li>
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamWriter.scala]: I would just format this similarly with https://github.com/apache/spark/blob/e06da95cd9423f55cdb154a2778b0bddf7be984c/sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala#L338-L366

**Comment by HyukjinKwon** [comment]
ping @niofire

**Comment by srowen** [comment]
This is fine, although unfortunately there are lots of instances of this type of formatting error. We don't need to fix them all, but what about looking for similar instances in `Column` and `Dataset` at least? I can see a few.

**Comment by srowen** [comment]
Ping @niofire to update or close

**Comment by niofire** [comment]
Sorry, was out of town, will update

**Comment by niofire** [comment]
@srowen Guessing you're referring to some other files? Could you link?

**Comment by srowen** [comment]
Have a look at `Column.scala` and `Dataset.scala` in `org.apache.spark.sql`. But, on a second look, this is how I see the lists render:

screen shot 2018-10-10 at 08 35 26

That already seems fine. Are you looking at javadoc for these classes? scaladoc looks OK.

**Comment by niofire** [comment]
From https://spark.apache.org/docs/2.3.2/api/java/org/apache/spark/sql/streaming/DataStreamWriter.html
image

I see java in that URL, is that actually referring to the java API?

**Comment by srowen** [comment]
Although it's the Scala API, it's callable from Java just as well. There's no Java-specific API here. So, yeah, actually it makes sense to have javadoc and scaladoc for this. And I think there are probably many markdown-like things that work in scaladoc not in javadoc. While you're welcome to fix all of it ideally, it may be way too much. OK, I'd say go ahead but at least fix all the instances of bullet lists in the .sql package. It's easy enough to search for something like `\*\s+-` as a regex to spot instances.

**Comment by niofire** [comment]
Seems like this wasn't able to build due to a transient error with CI

**Comment by niofire** [comment]
org.apache.spark.sql.hive.client.HiveClientSuites.(It is not a test it is a sbt.testing.SuiteSelector)
-> Unable to instantiate org.apache.hadoop.hive.metastore.HiveMetaStoreClient

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by srowen** [comment]
Hm, that's a weird error. Big javadoc failures from unrelated classes. This looks like errors you get when you run javadoc on translated Scala classes. No idea why it's just popped up. Let me run again.

```
[error] /home/jenkins/workspace/SparkPullRequestBuilder/core/target/java/org/apache/spark/serializer/SerializationDebugger.java:159: error: cannot find symbol
[error]   static private  org.apache.spark.serializer.SerializationDebugger.ObjectStreamClassReflection reflect ()  { throw new RuntimeException(); }
[error]                                                                    ^
[error]   symbol:   class ObjectStreamClassReflection
[error]   location: class SerializationDebugger
[error] /home/jenkins/workspace/SparkPullRequestBuilder/core/target/java/org/apache/spark/serializer/SerializationDebugger.java:22: error: class SerializationDebugger is already defined in package org.apache.spark.serializer
[error]   static private  class SerializationDebugger {
[error]                   ^
[error] /home/jenkins/workspace/SparkPullRequestBuilder/core/target/java/org/apache/spark/util/UtilsSuite.java:4: error: modifier static not allowed here
[error]     static public  class MalformedClass {
[error]                    ^
```

**Comment by niofire** [comment]
Is this happening for other PRs as well?

**Comment by HyukjinKwon** [comment]
Usually no.. i only this once in Tdas's PR before - at that time the order of methods in the change was related

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala [hunk 2]
```diff
@@ -46,11 +46,13 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
   private val df = ds.toDF()
 
   /**
+   * <ul>
    * Specifies the behavior when data or table already exists. Options include:
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala]: This one looks wrongly formatted btw. Looks ul should be below of this line

Comment (niofire) [reply sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala]: Crap yeah, will move it down

**Comment by HyukjinKwon** [comment]
Also, let's mention this PR targets to fix javadoc in the PR description, title and/or JIRA.

**Comment by HyukjinKwon** [comment]
Also, let's mention this PR targets to fix javadoc in the PR description and/or title.

**Comment by srowen** [comment]
Merged to master/2.4

**Pull request closed without merging**